### ISSUE TITLE
[BUG] Using @section('plugins.BootstrapSwitch', true) failed to turn on plugin

### ISSUE THREAD

**Issue by PiousVenom** [jeroennoten/Laravel-AdminLTE#902]
### Describe the bug
Following the instruction on [this page](https://github.com/jeroennoten/Laravel-AdminLTE/wiki/Advanced-Forms-Components#inputswitch), the plugin never loaded the JS file. 

### Steps To Reproduce
1. Run command: php artisan adminlte:plugins install --plugin=bootstrapSwitch
2. Add plugin information to the config/adminlte.php file
3. Added @section('plugins.BootstrapSwitch', true) to my blade
4. Added <x-adminlte-input-switch name="is_handheld" data-on-text="YES" data-off-text="NO" data-on-color="teal" checked/> to my code
5. Load page in Browser
6. Open browser console logs and see that there's an error: Uncaught TypeError: $(...).bootstrapSwitch is not a function

### Expected behavior
Plugin is turned on successfully when added to the blade

### Environment
Complete the next environment information.

| Item    | Version
| -------  | -------
| Laravel     | 8.42.1
|AdminLTE | 3.6
| OS           |Windows 10

### Additional context
If I add the following to my blade:

@section('js')
    <script src="{{asset('vendor/bootstrap-switch/js/bootstrap-switch.min.js')}}" type="text/javascript"></script>
@stop

Then everything works as expected.

**Comment by Shidersz** [comment]
@PiousVenom please share the plugins configuration you have on `config/admilte.php` and share the result of executing `php artisan adminlte:plugins`. Also, check if `bootstrap-switch.min.js` is included on the body when you use `@section('plugins.BootstrapSwitch', true)`.

**Comment by PiousVenom** [comment]
Plugins:

    'plugins' => [
        'Datatables' => [
            'active' => false,
            'files' => [
                [
                    'type' => 'js',
                    'asset' => false,
                    'location' => '//cdn.datatables.net/1.10.19/js/jquery.dataTables.min.js',
                ],
                [
                    'type' => 'js',
                    'asset' => false,
                    'location' => '//cdn.datatables.net/1.10.19/js/dataTables.bootstrap4.min.js',
                ],
                [
                    'type' => 'css',
                    'asset' => false,
                    'location' => '//cdn.datatables.net/1.10.19/css/dataTables.bootstrap4.min.css',
                ],
            ],
        ],
        'Select2' => [
            'active' => false,
            'files' => [
                [
                    'type' => 'js',
                    'asset' => false,
                    'location' => '//cdnjs.cloudflare.com/ajax/libs/select2/4.0.3/js/select2.min.js',
                ],
                [
                    'type' => 'css',
                    'asset' => false,
                    'location' => '//cdnjs.cloudflare.com/ajax/libs/select2/4.0.3/css/select2.css',
                ],
            ],
        ],
        'Chartjs' => [
            'active' => false,
            'files' => [
                [
                    'type' => 'js',
                    'asset' => false,
                    'location' => '//cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.0/Chart.bundle.min.js',
                ],
            ],
        ],
        'Sweetalert2' => [
            'active' => false,
            'files' => [
                [
                    'type' => 'js',
                    'asset' => false,
                    'location' => '//cdn.jsdelivr.net/npm/sweetalert2@8',
                ],
            ],
        ],
        'Pace' => [
            'active' => false,
            'files' => [
                [
                    'type' => 'css',
                    'asset' => false,
                    'location' => '//cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/blue/pace-theme-center-radar.min.css',
                ],
                [
                    'type' => 'js',
                    'asset' => false,
                    'location' => '//cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/pace.min.js',
                ],
            ],
        ],
        'BootstrapSwitch' => [
            'active' => false,
            'files' => [
                [
                    'type' => 'js',
                    'asset' => true,
                    'location' => 'vendor/bootstrap-switch/js/bootstrap-switch.min.js',
                ],
            ],
        ],
    ],

Adminlte:Plugins

+-------------------------------------+-------------------------+---------------+
| Plugin Name                         | Plugin Key              | Status        |
+-------------------------------------+-------------------------+---------------+
| Bootstrap4 Dual Listbox             | bootstrap4DualListbox   | Not Installed |
| Bootstrap Colorpicker               | bootstrapColorpicker    | Not Installed |
| Bootstrap Slider                    | bootstrapSlider         | Not Installed |
| Bootstrap Switch                    | bootstrapSwitch         | Installed     |
| bs-custom-file-input                | bsCustomFileInput       | Not Installed |
| Chart.js                            | chartJs                 | Not Installed |
| Datatables                          | datatables              | Not Installed |
| Datatables Plugins                  | datatablesPlugins       | Not Installed |
| Date Range Picker                   | daterangepicker         | Not Installed |
| Ekko Lightbox                       | ekkoLightbox            | Not Installed |
| FastClick                           | fastclick               | Not Installed |
| Filterizr                           | filterizr               | Not Installed |
| Flag Icon Css                       | flagIconCss             | Not Installed |
| Flot                                | flot                    | Not Installed |
| Fullcalendar                        | fullcalendar            | Not Installed |
| iCheck Bootstrap                    | icheckBootstrap         | Not Installed |
| InputMask                           | inputmask               | Not Installed |
| Ion.RangeSlider                     | ionRangslider           | Not Installed |
| jQuery Knob                         | jqueryKnob              | Not Installed |
| jQuery Mapael                       | jqueryMapael            | Not Installed |
| jQuery UI                           | jqueryUi                | Not Installed |
| jQuery Validation                   | jqueryValidation        | Not Installed |
| jQVMap                              | jqvmap                  | Not Installed |
| jsGrid                              | jsgrid                  | Not Installed |
| Pace Progress                       | paceProgress            | Not Installed |
| Select 2 with Bootstrap 4 Theme     | select2                 | Not Installed |
| Sparklines                          | sparklines              | Not Installed |
| Summernote                          | summernote              | Not Installed |
| Sweetalert 2 with Bootstrap 4 Theme | sweetalert2             | Not Installed |
| Tempus Dominus for Bootstrap 4      | tempusdominusBootstrap4 | Not Installed |
| Toastr                              | toastr                  | Not Installed |
+-------------------------------------+-------------------------+---------------+

Here's screenshots of the <body> and <head>
![body](https://user-images.githubusercontent.com/4548799/119270211-34920880-bbc1-11eb-92f7-1d9005552163.PNG)
![head](https://user-images.githubusercontent.com/4548799/119270213-38258f80-bbc1-11eb-96bb-21fddd9f168b.PNG)

**Comment by Shidersz** [comment]
@PiousVenom That's rare. The javascript file is not included in the body. Can you try changing `'active' => false` to `'active' => true` and check if the `vendor/bootstrap-switch/js/bootstrap-switch.min.js'` file is included within the body?

**Comment by PiousVenom** [comment]
1.) No, I never published the views.
2.) I just updated to 3.6. Had 3.5 prior.

I've tried setting it `'active' => true` in the plugin config, but it still never loaded.

**Comment by Shidersz** [comment]
@PiousVenom Not sure what may be the problem. I have the same setup but in a Linux machine and all is working good. Just to be sure, check you don't have a `resources/views/vendor/adminlte` folder created. That one is created when you publish the views. On the other hand, are you are using **laravel mix** enabled on the configuration option?

**Comment by PiousVenom** [comment]
@Shidersz I've double checked and I don't have any vendor folders in my `resources/views`. And the config has `'enabled_laravel_mix' => false,`. Just not sure what else to do, aside from attempting to start over. But ugh at that.

**Comment by Shidersz** [comment]
Have you follow the documented [update procedure](https://github.com/jeroennoten/Laravel-AdminLTE/wiki/Updating)?

**Comment by PiousVenom** [comment]
I followed all the steps, and nothing was updated. However, I did `php artisan adminlte:install --only=config`, and sure enough, that fixed it. I honestly didn't think to do that step when I first updated from 3.5 to 3.6.

Thank you for your patience and your help.

**Comment by Shidersz** [comment]
That command will just rewrite the `config` file with the package default one. Note you may loss all the changes you have pushed on the config file after that. So the problem is solved after pushing the default package configuration file?

**Comment by PiousVenom** [comment]
I made a backup of my current and am going through and updating with the work I'd added to it before. But yes, problem was solved. Thank you, again. Appreciate ya.

**Comment by Shidersz** [comment]
Ok, I close this one then...

**Comment by miobit** [comment]
> 
> 
> I followed all the steps, and nothing was updated. However, I did `php artisan adminlte:install --only=config`, and sure enough, that fixed it. I honestly didn't think to do that step when I first updated from 3.5 to 3.6.
> 
> Thank you for your patience and your help.

Hi, i had same issue but with different plugin Select2 - i wanted to add dependencies from local storage:
> $(...).select2 is not a function 

TL;DR; i have to use cdn to go, but your tip helped `php artisan adminlte:install --only=config` , THX guys

i tried:
- 'enabled_laravel_mix' => false,
- plugin active true/false
- asset active true/false
- 'location' => '../vendor/select2/js/select2.full.min.js', (of course with  ./  ../ ../../  )
- separate installation of select2 and theme with bootstrap theme

i ended with publishing adminlte views and search for issue - plugins blade is not used at all - so no file was added to DOM.
manually added files to  select2 blade in exported views made success.. 

of course i can load plugins manually but ... yk

**Comment by Shidersz** [comment]
@miobit The **plugins blade** is used from the **master blade**. You may try debugging the plugins blade file, for example:

```blade
{{ logger('Plugins file hit!') }}

@foreach(config('adminlte.plugins') as $pluginName => $plugin)

    {{ logger('Loading plugin: ' . $plugin['name'] ?? $pluginName) }}

    @if($plugin['active'] || View::getSection('plugins.' . ($plugin['name'] ?? $pluginName)))
        @foreach($plugin['files'] as $file)

            {{-- Setup the file location  --}}
            @php
                if (! empty($file['asset'])) {
                    $file['location'] = asset($file['location']);
                }
            @endphp

            {{-- Check requested file type --}}
            @if($file['type'] == $type && $type == 'css')
                <link rel="stylesheet" href="{{ $file['location'] }}">
            @elseif($file['type'] == $type && $type == 'js')
                <script src="{{ $file['location'] }}" @if(! empty($file['defer'])) defer @endif></script>
            @endif

        @endforeach
    @endif
@endforeach
```

And then check the laravel log file to see if the file is included (first log) and if there exists a plugin on the configuration file. You may have an error on the configuration file, that may result on undefined `adminlte.plugins` configuration.

Other problem may be that you have executed the artisan command to cache the laravel configuration, and now any change you made on any configuration will not be visible. In this case, try to clean the configuration cache with:

```sh
php artisan config:clear
```

**Comment by miobit** [comment]
Thank you for fast response.

This issue was probably created by misconfigured conf file. After replacing it everything works fine

I tried debug but wasn't triggered from plugins blade (this file wasn't executed. No error in error log apache/laravel)
Also I had 
`php artisan optimize:clear ` and for sure config:clear
